new-log-viewer/src/contexts/StateContextProvider.tsx

```diff
@@ -240,22 +210,79 @@ const StateContextProvider = ({children}: StateContextProviderProps) => {
         handleMainWorkerResp,
     ]);
 
-    const loadPage = (newPageNum: number) => {
+    const loadPageByCursor = useCallback((
+        cursor: CursorType,
+    ) => {
         if (null === mainWorkerRef.current) {
             console.error("Unexpected null mainWorkerRef.current");
 
             return;
         }
+
         workerPostReq(mainWorkerRef.current, WORKER_REQ_CODE.LOAD_PAGE, {
-            cursor: {code: CURSOR_CODE.PAGE_NUM, args: {pageNum: newPageNum}},
+            cursor: cursor,
             decoderOptions: getConfig(CONFIG_KEY.DECODER_OPTIONS),
         });
-    };
+    }, []);
+
+    const getPageNumCursor = useCallback((navAction: NavigationAction): Nullable<CursorType> => {
```

Comment:
Yeah, I know I asked for it (partly because we were also having to export STATE_DEFAULT iirc) but looking at the code now, I feel like adding more `useCallbacks` into `StateContextProvider` is making the code more confusing. Would there be a negative effect to just moving it outside of `StateContextProvider` but keeping it in the same file?